src/Database/DataFeed.php

```diff
@@ -211,9 +217,15 @@ protected function processCollection()
             /*
              * Flush the select, add ID and tag
              */
-            $cleanQuery = $cleanQuery->select(Db::raw($keyName." as id"));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT '".$tag."') as ".$this->tagVar));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT ".$sorting.") as ".$this->sortVar));
+
+            $cast = $this->getDbDialect() === 'pgsql' ? 'text' : 'binary';
+            $idSelect = sprintf("%s as id", $keyName);
+            $tagSelect = sprintf("cast('%s' as %s) as %s", $tag, $cast, $this->tagVar);
+            $sortSelect = sprintf("%s as %s", $sorting, $this->sortVar);
+
+            $cleanQuery = $cleanQuery->select(Db::raw($idSelect));
+            $cleanQuery = $cleanQuery->addSelect(Db::raw($tagSelect));
```

Comment:
@dimensi Could this line and the line below instead be chained to the previous `$cleanQuery->select()` call, as opposed to overwriting the `$cleanQuery` query 3 times?

Reply:
ok